packages/world/src/utils/InventoryUtils.sol

```diff
@@ -474,10 +436,18 @@ library InventoryUtils {
     }
 
     uint16 maxSlots = EntityObjectType._get(owner).getMaxInventorySlots();
-    require(occupiedIndex < maxSlots, "All slots used");
-    InventorySlot._setOccupiedIndex(owner, occupiedIndex, occupiedIndex);
-    Inventory._push(owner, occupiedIndex);
-    return occupiedIndex;
+    for (uint16 i = 0; i < maxSlots; i++) {
```

Comment:
I wonder if we're looping like this we could go the simpler solution of not storing occuied index. Ie that idea we had before of Inventory just being stored in a single table

Reply:
Let's think through this. My main concern is that it could become really expensive to list through inventories for entities with max slots 2^16. Like on transfer all to air on death, you could get stuck without being able to die as for each of the player slots you would need to iterate through air slots until you find an empty slot or one with enough capacity for the object type.. are you thinking of something different?

Reply:
we could make air not have an inventory, but then pickup and drop would need to change to not be slot based, which maybe is fine

Reply:
will give it a go, hopefully it doesn't change the external api too much